packages/typegpu/src/name.ts

```diff
@@ -0,0 +1,53 @@
+import { $getNameForward } from './shared/symbols.ts';
+
+interface MetaData {
+  name: string | undefined;
+}
+
+declare global {
+  var __TYPEGPU_META__: WeakMap<object, MetaData>;
+}
```

Comment:
Declaring this here causes any project using TypeGPU to be aware of this value. It will complicate our internals a bit without this block of code, but it won't pollute the global types 🙇‍♂️

<img width="618" alt="image" src="https://github.com/user-attachments/assets/5e60cd7d-b786-4a33-991a-ec8762054a85" />


```suggestion
interface GlobalWithMeta {
  __TYPEGPU_META__: WeakMap<object, MetaData>;
}
```